api/traces/v1/http.go

```diff
@@ -163,6 +165,47 @@ func NewV2Handler(read *url.URL, opts ...HandlerOption) http.Handler {
 	return r
 }
 
+// Parse a URL; if the URL includes `{tenant}` that portion will be replaced by the tenant
+func ExpandTemplatedUpstream(templateUpstream, tenant string) (*url.URL, error) {
+	rawTracesReadEndpoint := strings.Replace(templateUpstream, "{tenant}", tenant, 1)
+	return url.ParseRequestURI(rawTracesReadEndpoint)
+}
+
+// middlewareSetTemplatedUpstream is a variation of proxy.MiddlewareSetUpstream()
+// with additional processing if the upstream includes "{tenant}"
```

Comment:
```suggestion
// with additional processing if the upstream includes "{tenant}".
```

Reply:
I seem to make this mistake a lot.  Perhaps Observatorium could add a linter for it? https://github.com/tetafro/godot